src/ekat/ekat_workspace_impl.hpp

```diff
@@ -124,6 +125,29 @@ void WorkspaceManager<T, D>::report() const
 #endif
 }
 
+template <typename T, typename D>
+void WorkspaceManager<T, D>::setup (int size, int max_used, TeamPolicy policy,
+                                    const double& overprov_factor)
+{
```

Comment:
Can the constructors call setup to reduce a bit of duplication?

Reply:
Agreed